src/Home/CreateRoomButton.js

```diff
@@ -1,18 +1,42 @@
+import { useEffect } from "react";
 import { supabase } from "../supabase_client";
 import { generateRoomCode } from "../utils";
 
 const CreateRoomButton = (props) => {
-  const { player, setPage, setRoomCode, setIsHost } = props;
+  const { player, setPage, setRoomCode, setIsHost, roomCode } = props;
+  let pageSelect = "home";
+  let hostSelect = false;
+  let roomSelect = null;
+
+  const fetchRoomCodes = async () => {
+    const { data } = await supabase.from("Players").select("roomID");
+    const registeredRooms = data;
+    return registeredRooms;
+  };
+
   const handleClick = async () => {
     const roomID = generateRoomCode();
-    setRoomCode(roomID);
-    const { data, error } = await supabase
-      .from("Players")
-      .update({ roomID: roomID })
-      .match({ name: player });
-    setIsHost(true);
-    setPage("lobby");
+    const registeredRooms = await fetchRoomCodes();
+    const existingRoomCode = registeredRooms.find(
+      (rooms) => rooms.roomID === roomID
+    );
+
+    if (existingRoomCode === undefined) {
+      const { data, error } = await supabase
+        .from("Players")
+        .update({ roomID: roomID })
+        .match({ name: player });
+      hostSelect = true;
+      roomSelect = roomID;
+      pageSelect = "lobby";
+    } else {
+      handleClick();
+    }
+    setIsHost(hostSelect);
+    setPage(pageSelect);
+    setRoomCode(roomSelect);
```

Comment:
Ln 33: `handleClick();`, the behaviour of this line is weird. Since its async it may or may not run after the rest function completed. If you write await in front of it, when it finishes its own run, it'll come back to this run and call lns 35-37 which you probably dont want. tldr the behaviour here is kinda wonky.

A better approach can be similar as the playername suggestion. Here we can let supabase tell us there was an uniqueness conflict, and if there was we generate a new room code and do another post request. I'd suggest wrapping just those two actions in a separate function so you can avoid the issue from the first paragraph.